support/cas-server-support-pm-webflow/src/main/java/org/apereo/cas/pm/web/flow/actions/PasswordChangeAction.java

```
@@ -79,8 +80,10 @@ protected Event doExecuteInternal(final RequestContext requestContext) {
         val clientInfo = ClientInfoHolder.getClientInfo();
 
         val bean = getPasswordChangeRequest(requestContext);
-        Optional.ofNullable(WebUtils.getCredential(requestContext, UsernamePasswordCredential.class))
-            .ifPresent(credential -> bean.setCurrentPassword(credential.getPassword()));
+        if (CasWebflowConfigurer.FLOW_ID_PASSWORD_RESET.equals(requestContext.getActiveFlow().getId())) {
+            Optional.ofNullable(WebUtils.getCredential(requestContext, UsernamePasswordCredential.class))
```

Can this credential here ever be null?

You may want to rewrite this simpler: 

```
var c = WebUtils.getCredential(requestContext, UsernamePasswordCredential.class)
bean.setCurrentPassword(c.getPassword())
```

---

> If you are receiving the current password from the page, you need to make sure it matches the current password for the user. I don't see where this validation is happening.

I use this password to call syncope endpoint rest to change password. If password is incorrect, http request fails

---

Right, this is exactly what I was saying before:

1. You must make sure the current password is required by the backend code, when it is in fact required and you're in the right flow.
2. And you must make sure the provided password matches the current password for the user that is available to the credential.

otherwise, I can manipulate the request to pass a blank password, and nothing in the backend is stopping me.

---

> I use this password to call syncope endpoint rest to change password. If password is incorrect, http request fails

Once more, this has to work regardless of the backend password store. It has to work for LDAP, JDBC, and anything else that today is not checking for or using the current password.

---

> otherwise, I can manipulate the request to pass a blank password, and nothing in the backend is stopping me.

Yes but if the password is incorrect, authentication for http request fails and update operation isn't done. It's good to make a check before this and i'll work for this, but also without check, if you try to insert bad current password, operation goes in error

---

> Once more, this has to work regardless of the backend password store. It has to work for LDAP, JDBC, and anything else that today is not checking for or using the current password.

Thus i have to add some check about current password also for this resource right?

---

> Yes but if the password is incorrect, authentication for http request fails 

1. This only works for Syncope; that is not the requirement here. This has to work for anything and everything.
2. If you know the password is wrong beforehand, it is unnecessary to make an HTTP call for Syncope and let it fail. Why make that call? That is going to cost.

---

> Thus i have to add some check about current password also for this resource right?

Correct. Your webflow action must be absolutely sure all things pass before it passes the op to the real password mgmt store (which may be syncope or anything else).